### PR TITLE
Collect mustgather from all clusters in case of multicluster scenario

### PR DESCRIPTION
**Pull request by shylesh**
This PR  addresses
1. Collects mustgather from all the clusters involed in multicluster scenario
2. Collects submariner logs in case of regional-dr deployments

Signed-off-by: Shylesh Kumar Mohan <shmohan@redhat.com>

### CODE REVIEW

**Comment by shylesh** [comment]
According to the discussion with @prsurve @keesturam we need a multithreaded mustgather which parallely gathers the logs from all the clusters at the same time.

**Comment by shylesh** [comment]
Rebased on latest master

**Comment by shylesh** [comment]
Rebased on latest master

**Comment by shylesh** [comment]
Rebased on latest master

**Comment by shylesh** [comment]
rebased on latest master

**Comment by shylesh** [comment]
Rebased on latest master

**Comment by shylesh** [comment]
Rebased and conflict resolved

**Comment by petr-balogh** [comment]
I see:
```
2023-03-09 14:08:00  13:07:55 - ThreadPoolExecutor-14_0 - ocs_ci.ocs.utils - INFO  - OCS logs will be placed in location /home/jenkins/current-cluster-dir/logs/failed_testcase_ocs_logs_1678364822/jnk-pr7022b4295/test_quickstart_presense[OpenShift Container Storage Configuration & Management]_ocs_logs/ocs_must_gather
2023-03-09 14:08:00  13:07:55 - ThreadPoolExecutor-14_0 - ocs_ci.utility.utils - INFO  - Executing command: oc --kubeconfig /home/jenkins/current-cluster-dir/openshift-cluster-dir/auth/kubeconfig adm must-gather --image=quay.io/rhceph-dev/ocs-must-gather:latest-4.12 --dest-dir=/home/jenkins/current-cluster-dir/logs/failed_testcase_ocs_logs_1678364822/jnk-pr7022b4295/test_quickstart_presense[OpenShift Container Storage Configuration & Management]_ocs_logs/ocs_must_gather
2023-03-09 14:08:07  13:08:06 - MainThread - ocs_ci.ocs.utils - INFO  - None
2023-03-09 14:08:07  FAILED
```

But here in this location:
https://url.corp.redhat.com/b2375f4

I see empty ocs_must_gather directories for failed test cases which is very concerning me.

**Comment by shylesh** [comment]
resolved conflict and rebased on latest master

File: ocs_ci/ocs/constants.py [hunk 1]
```diff
@@ -1928,6 +1928,7 @@
 ACM_CLUSTER_DEPLOYMENT_LABEL_KEY = "hive.openshift.io/cluster-deployment-name"
 ACM_CLUSTER_DEPLOYMENT_SECRET_TYPE_LABEL_KEY = "hive.openshift.io/secret-type"
 ACM_4_7_MULTICLUSTER_URL = "/multicloud/infrastructure/clusters/managed"
+ACM_CSV_LABEL = "operators.coreos.com/advanced-cluster-management.open-cluster-management= -n open-cluster-management"
```
Comment (prsurve) [inline ocs_ci/ocs/constants.py:1931]: -n should be removed

File: ocs_ci/ocs/utils.py [hunk 2]
```diff
@@ -1068,25 +1084,96 @@ def collect_ocs_logs(dir_name, ocp=True, ocs=True, mcg=False, status_failure=Tru
...
                 sleep(30)
+    # Collect ACM logs only from ACM
+    if cluster_config.ENV_DATA.get("multicluster_mode", None) == "regional-dr":
+        if cluster_config.MULTICLUSTER.get("acm_cluster", False):
+            image_prefix = "acm_must_gather"
+            acm_mustgather_path = os.path.join(
+                os.path.expanduser(cluster_config.RUN["log_dir"]),
+                f"{dir_name}_{cluster_config.RUN['run_id']}",
+            )
+            csv_cmd = f"oc get csv -l {constants.ACM_CSV_LABEL} -o json"
```
Comment (prsurve) [inline ocs_ci/ocs/utils.py:1117]: -n namespace should be passed

File: ocs_ci/ocs/constants.py (hunk 1 above)
Comment (shylesh) [reply ocs_ci/ocs/constants.py]: done

File: ocs_ci/ocs/utils.py (hunk 2 above)
Comment (shylesh) [reply ocs_ci/ocs/utils.py]: done

**Comment by shylesh** [comment]
Rebased on latest master

**Comment by shylesh** [comment]
Rebased on latest master

**Comment by shylesh** [comment]
Verified: https://url.corp.redhat.com/4-13

**Review by petr-balogh — APPROVED** [review]

**Review by prsurve — APPROVED** [review]

**Comment by shylesh** [comment]
Looks like regression run has must gather collection issue, failed tests doesn't have must gather collected: https://url.corp.redhat.com/regression1

**Review by am-agrawa — DISMISSED** [review]

**Comment by shylesh** [comment]
Rebased on latest master

**Comment by shylesh** [comment]
https://url.corp.redhat.com/prverification and regression: https://url.corp.redhat.com/regression2

**Review by prsurve — APPROVED** [review]

File: ocs_ci/ocs/utils.py [hunk 3]
```diff
@@ -1070,25 +1090,108 @@ def collect_ocs_logs(dir_name, ocp=True, ocs=True, mcg=False, status_failure=Tru
...
+        submariner_log_path = os.path.join(
+            log_dir_path,
+            "submariner",
+        )
+        run_cmd(f"mkdir -p {submariner_log_path}")
+        cwd = os.getcwd()
+        run_cmd(f"chmod -R 777 {submariner_log_path}")
+        os.chdir(submariner_log_path)
+        submariner_log_collect = (
+            f"subctl gather --kubeconfig {cluster_config.RUN['kubeconfig']}"
```
[the remark this answers is not in the document]
Comment (am-agrawa) [reply ocs_ci/ocs/utils.py:1150]: As discussed today, we will take other logs collection for sumbariner in a different PR.

**Review by petr-balogh — APPROVED** [review]

**Pull request merged**